src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.hpp

```diff
@@ -28,6 +28,10 @@ class RefTransposeExecutorBuilder : public TransposeExecutorBuilder {
     bool isSupported(const TransposeParams& transposeParams,
                      const std::vector<MemoryDescPtr>& srcDescs,
                      const std::vector<MemoryDescPtr>& dstDescs) const override {
+        // only ncsp layout is currently supported
+        if (!srcDescs[0]->hasLayoutType(LayoutType::ncsp))
```

Comment:
ref transpose impl looks pretty generic: it [utilizes](https://github.com/openvinotoolkit/openvino/blob/574a642c74689c0fdee63003012bfafbae60df57/src/plugins/intel_cpu/src/nodes/executors/common/ref_transpose.cpp#L59-L60) strides to get correct element offset. Stride itself already [accomodates ](https://github.com/openvinotoolkit/openvino/blob/574a642c74689c0fdee63003012bfafbae60df57/src/plugins/intel_cpu/src/memory_desc/cpu_blocked_memory_desc.cpp#L51-L62) layout details based on order field. In other words I would expect ref impl to fully support non-blocked layouts.

Reply:
you're right, the problem was in incorrectly passed transposed parameters.